roles/sssd/templates/ldap_backup.sh.j2

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+_ldap_credentials="/etc/openldap/readonly-ldapsearch-credentials.bash"
+source ${_ldap_credentials}
+
+_ldap_dest="/root/ldap_backup"
+test -e "${_ldap_credentials}" || exit 1
+
+_daily_keep=31     # one month worth
+_weekly_keep=52    # one year worth
+
+for _each_ldap in "${domain_names[@]}"; do
+   echo "> running backup for ${_each_ldap} <"
+   test -e "${_ldap_dest}/${_each_ldap}" || mkdir "${_ldap_dest}/${_each_ldap}"
+   for _scope in person group; do
+      echo " - running backup for ${_scope}"
+      _scope_destination="${_ldap_dest}/${_each_ldap}/${_scope}"
+      test -e "${_scope_destination}" || mkdir "${_scope_destination}"
+      /usr/bin/ldapsearch -H "${domain_configs[${_each_ldap}'_uri']}" -D"${domain_configs[${_each_ldap}'_bind_dn']}" -b"${domain_configs[${_each_ldap}'_search_base']}" -w"${domain_configs[${_each_ldap}'_bind_pw']}" "(objectClass=person)" -LLL | gzip > ${_scope_destination}/$(date +%Y-%m-%dT%H-%M-%S).ldif.gz
+      cd "${_scope_destination}" || { logger -t ldap_backup "${0}: Error, problem with changing dir to ${_scope_destination}"; exit 1; }
+      ## Check if this weeks backup has been made and if not, make it
+      ##
+      if ! ls ${_scope_destination}/*-weekly-$(date +%Y%U).ldif.gz 1>/dev/null 2>&1; then
+         # Make a copy of the _latest backup as a weekly backup
+         _latest=$(ls -1 ${_scope_destination}/*.ldif.gz | tail -n 1)
```

Comment:
This `ls` + `tail` depends on the (sort) order in which `ls` returns results: that can be dangerous.... We may need a different construct or add explicit env vars like 
```
export LC_COLLATE=....
export LC_TYPE=....
export LC_ALL=....
```
at the top of the script and then do an explicit sort of the ```ls``` results to make sure the order is as expected.

Reply:
Usually we do not change the default `LC_` values for root - so I believed it to be always set to `C` (or `C.utf8`).
But I can also hardcode it to `LC_ALL=C`